[matthewyip1511] iP

matthewyip1511 opened the pull request:
## PoChat

> “PoChat is really useful. It really is” – Matthew Yip 

PoChat frees your mind from having to remember things you need to do. It's:

- **Text-based**
- **Easy to learn**
- **~~FAST~~, SUPER FAST** to use

All you need to do is:

1. Download it from [here].
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉.

And it is **FREE**!

### Features:
- Managing tasks
- Managing deadlines
- Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(PoChat, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Set up level-0

Commit:
Set up level-0

Commit:
Set up level 1

Commit:
Set up Level 2

Commit:
Set up level 3

Commit:
Set up level 4

Commit:
Set up level 4

Commit:
"Set up A-TextUiTesting"

Commit:
"Set up A-TextUiTesting"

Commit:
Set up Level-6

Commit:
Set up level 7 to save data to hard disk

Commit:
Merge branch 'branch-level-7' after implementing save

Commit:
Set up Level 8 datetime

Commit:
Merge branch 'branch-level-8' after implementing datetime

Commit:
Set up A-MoreOOP

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Set up A-Packages

Commit:
Merge branch 'add-gradle-support' to start level A-Gradle

Commit:
Set up gradle in ip

Commit:
Set up unit tests under A-JUnit

Commit:
Set up unit tests under A-JUnit

Commit:
Edit build.gradle to support creating jar files with the poChat base name

Commit:
Add JavaDoc comments to all public methods

Commit:
Merge branch 'master' into branch-A-JavaDoc to add javadocs

Commit:
Rename variables in Parser sayGoodbye and add javadoc for it

Commit:
Set up level 9

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Fix merge conflict

Commit:
Add back function lost while fixing merge conflict

jason-chuah commented on src/main/java/poChat/bot/Parser.java:
```diff
@@ -0,0 +1,167 @@
+        this.addTaskToList(task);
+    }
+
+    private void replyWithListOfTextsEntered() {
```
Maybe you can consider shortening this to replyWithListOfTexts or replyWithList? So that it's slightly shorter

jason-chuah commented on src/main/java/poChat/exceptions/ChatHistoryFileMissingException.java:
```diff
@@ -0,0 +1,8 @@
+package poChat.exceptions;
+
+public class ChatHistoryFileMissingException extends RuntimeException {
```
Maybe you can consider shortening this to FIleMissingException?

jason-chuah commented on src/main/java/poChat/bot/ChatData.java:
```diff
@@ -0,0 +1,56 @@
+package poChat.bot;
+
+import java.io.*;
```
wildcard import

jason-chuah commented on src/main/java/poChat/tasks/Task.java:
```diff
@@ -0,0 +1,91 @@
+        this.isDone = isDone;
+    }
+
+    private static String formatLoadedDateTime(String dateTime) {
```
could possibly be shortened to formatDateTime?

jason-chuah reviewed (commented):
LGTM. Just feel that a few variable names are unnecessarily long, especially the ones I highlighted.

JYL27 commented on src/main/java/poChat/bot/Parser.java:
```diff
@@ -0,0 +1,167 @@
+     * <code>false</code> which means the conversation continues
+     */
+
+    public boolean parse(String textInput) throws TaskDescriptionEmptyException {
```
Boolean methods should be named to sound like booleans. I've noticed this issue in the PoChat class as well.

JYL27 commented on src/main/java/poChat/bot/Parser.java:
```diff
@@ -0,0 +1,167 @@
+import poChat.tasks.Task;
+import poChat.tasks.ToDo;
+
+class Parser {
```
I like how descriptive the method names are in general👍.

JYL27 commented on src/main/java/poChat/bot/Parser.java:
```diff
@@ -0,0 +1,167 @@
+    }
+
+    private void replyToInvalidInput() {
+        String errorMessage = "Please enter a valid input and try again! Some examples of valid inputs are:\n" +
```
Breaks here should come before the plus operators. I've noticed this issue in the PoChat and Event classes as well.

JYL27 commented on src/main/java/poChat/tasks/Task.java:
```diff
@@ -0,0 +1,91 @@
+            return new ToDo(taskDescription, isDone);
+
+        } else if (taskObjectString.startsWith("[D]")) {
+            boolean isDone = taskObjectString.charAt(4) == 'X';
```
I like the block separations here, makes the code much more readable👍.

JYL27 reviewed (commented):
There are some violations of the coding standards, namely breaks, boolean method naming and wildcard import. Looks good otherwise.

Commit:
Add checkstyle to project

Quasant commented on src/main/java/pochat/exceptions/ChatHistoryFileMissingException.java:
```diff
@@ -0,0 +1,8 @@
+public class ChatHistoryFileMissingException extends RuntimeException {
+    public ChatHistoryFileMissingException() {
+        super("The chat history file does not exist! If this is your first time running the chatbot, please" +
+                " refer to the readme on how to initialise the chat_data.txt file for the first time");
```
Small nit - to break before the `+` operator

Quasant commented on src/main/java/pochat/bot/Parser.java:
```diff
@@ -0,0 +1,167 @@
+     * <code>false</code> which means the conversation continues
+     */
+
+    public boolean parse(String textInput) throws TaskDescriptionEmptyException {
```
Perhaps consider adding a @throws in the java docs for the instances when this exception will occur.

Quasant commented on src/main/java/pochat/bot/Parser.java:
```diff
@@ -0,0 +1,167 @@
+        return isChatOver;
+    }
+
+    /**
+     * Prints the goodbye message to the user
+     */
+    private void sayGoodbye() {
```
Not a violation of coding standards, but for this private method, the method was named well and is self-explanatory.

Quasant reviewed (commented):
Overall, LGTM. I think the most common improvement would be with regards to line breaks, where you break after the + operator. 

Breaking before the + operator makes all of them aligned which could make the code more aesthetically pleasing.

Commit:
Change pochat.tasks to tasks which was a typo

Commit:
Set up GUI for poChat chatbot

Commit:
Merge branch 'branch-level-10' to add GUI to chatbot

Commit:
Add assertions to code
Assertions are in the Parser.java class to check that tasks are correctly added and removed from the list
Assertions are also added in the Task class to check that files are correctly marked as done or not done in their corresponding functions

Commit:
Fix coding standard violations
Coding standard violations were fix by going through the error messages generated by CheckStyle
Main issues were the lack of javdoc comments as well as wrong levels of indentation or wrong orders of import statements

Commit:
Refactor code to improve code quality
Chanegs made include to rename all reply functions in the parser to be consistent and more descriptive
Magic literals are also removed in the Parser and replaced with variables with descriptive names
while(true) loop also removed in ChatData and replaced with more descriptive boolean checking

Commit:
Add back hello feature that was left out when moving to gui

matthewyip1511 committed:
Merge pull request #1 from matthewyip1511/branch-A-CodeQuality

Merge branch-A-CodeQuality into master

Commit:
Fix merge conflict and minor refactoring to rename function names

Commit:
Fix merge conflict

Commit:
Fix merge conflict

Commit:
Fix merge conflict

Commit:
Fix merge conflict

matthewyip1511 committed:
Merge pull request #2 from matthewyip1511/branch-A-Assertions

Add assertions to code

Commit:
Merge branch-A-Assertions into Master

Commit:
Rename functions in Parser.java

Commit:
Add C-DetectDuplicates support to detect duplicates when adding tasks

Commit:
Fix checkStyle violations

matthewyip1511 committed:
Merge pull request #3 from matthewyip1511/branch-BCD-extension

Merge BCD-extension into master

Commit:
Add ui class to handle return messages to user

Commit:
Add more SLAP and reduce long methods

matthewyip1511 committed:
Merge pull request #5 from matthewyip1511/branch-add-ui

Add UI class to support return messages

Commit:
Change button color to green

Commit:
Update README for user guide

Commit:
Update README on landing page

Commit:
Update README for user guide to add error handling

Commit:
Update README to add bold text

Commit:
Update README to add Ui screenshot at the top

Commit:
Remove sayHello in PoChat.java that is no longer needed

Commit:
Refactor Task.java

Commit:
Resize images in GUI and refactor ui message for invalid input

Commit:
Update load chat data to create the file if it does not exist

Commit:
Refactor to remove magic literals and add javadocs

Commit:
Refactor to remove magic literals and add javadocs

Commit:
Add more Junit test

Commit:
Update README

Commit:
Update README

Commit:
Add support for detecting out of range task index for mark, unmark and delete commands

Commit:
Add support for detecting out of range task index for mark, unmark and delete commands

Commit:
Add support to catch invalid todo, deadline and event

Commit:
Add error handling for task index that is not an integer

Commit:
Add error handling to delete as well

Commit:
Fix bug related to parsing remove and unmark

Commit:
Update README on main page

Commit:
Update README

Commit:
Refactor Parser.java
